targets/TARGET_STM/i2c_api.c

```diff
@@ -440,963 +549,1134 @@ void i2c_init_direct(i2c_t *obj, const i2c_pinmap_t *pinmap)
+		}
 #elif defined(I2C_IP_VERSION_V2)
-        HAL_I2C_Master_Seq_Transmit_IT(handle, address, (uint8_t *)tx, tx_length, I2C_FIRST_FRAME);
+		HAL_I2C_Master_Seq_Transmit_IT(handle, address, (uint8_t *)tx, tx_length, I2C_FIRST_FRAME);
 #endif
-    }
+	}
 }
 
 
 uint32_t i2c_irq_handler_asynch(i2c_t *obj)
 {
 
-    struct i2c_s *obj_s = I2C_S(obj);
-    I2C_HandleTypeDef *handle = &(obj_s->handle);
+	struct i2c_s *obj_s = I2C_S(obj);
+	I2C_HandleTypeDef *handle = &(obj_s->handle);
 
-    HAL_I2C_EV_IRQHandler(handle);
-    HAL_I2C_ER_IRQHandler(handle);
+	HAL_I2C_EV_IRQHandler(handle);
+	HAL_I2C_ER_IRQHandler(handle);
 
-    /*  Return I2C event status */
-    return (obj_s->event & obj_s->available_events);
+	/*	Return I2C event status */
+	return (obj_s->event & obj_s->available_events);
 }
 
 uint8_t i2c_active(i2c_t *obj)
 {
 
-    struct i2c_s *obj_s = I2C_S(obj);
-    I2C_HandleTypeDef *handle = &(obj_s->handle);
+	struct i2c_s *obj_s = I2C_S(obj);
+	I2C_HandleTypeDef *handle = &(obj_s->handle);
 
-    if (handle->State == HAL_I2C_STATE_READY) {
-        return 0;
-    } else {
-        return 1;
-    }
+	if (handle->State == HAL_I2C_STATE_READY) {
+		return 0;
+	} else {
+		return 1;
+	}
 }
 
 void i2c_abort_asynch(i2c_t *obj)
 {
 
-    struct i2c_s *obj_s = I2C_S(obj);
-    I2C_HandleTypeDef *handle = &(obj_s->handle);
+	struct i2c_s *obj_s = I2C_S(obj);
+	I2C_HandleTypeDef *handle = &(obj_s->handle);
 
-    /* Abort HAL requires DevAddress, but is not used. Use Dummy */
-    uint16_t Dummy_DevAddress = 0x00;
+	/* Abort HAL requires DevAddress, but is not used. Use Dummy */
+	uint16_t Dummy_DevAddress = 0x00;
 
-    HAL_I2C_Master_Abort_IT(handle, Dummy_DevAddress);
+	HAL_I2C_Master_Abort_IT(handle, Dummy_DevAddress);
+}
+
+/**
+  * @brief	Calculate SCLL and SCLH and find best configuration.
+  * @param	clock_src_freq I2C source clock in HZ.
+  * @param	I2C_speed I2C frequency (index).
+  * @retval config index (0 to I2C_VALID_TIMING_NBR], 0xFFFFFFFF for no valid config.
+  */
+uint32_t I2C_Compute_SCLL_SCLH(uint32_t clock_src_freq, uint32_t I2C_speed)
+{
+	uint32_t ret = 0xFFFFFFFFU;
+	uint32_t ti2cclk;
+	uint32_t ti2cspeed;
+	uint32_t prev_error;
+	uint32_t dnf_delay;
+	uint32_t clk_min, clk_max;
+	uint32_t scll, sclh;
+	uint32_t tafdel_min;
+
+	ti2cclk	  = (SEC2NSEC + (clock_src_freq / 2U)) / clock_src_freq;
+	ti2cspeed	= (SEC2NSEC + (I2C_Charac[I2C_speed].freq / 2U)) / I2C_Charac[I2C_speed].freq;
+
+	tafdel_min = (I2C_USE_ANALOG_FILTER == 1U) ? I2C_ANALOG_FILTER_DELAY_MIN : 0U;
+
+	/* tDNF = DNF x tI2CCLK */
+	dnf_delay = I2C_Charac[I2C_speed].dnf * ti2cclk;
+
+	clk_max = SEC2NSEC / I2C_Charac[I2C_speed].freq_min;
+	clk_min = SEC2NSEC / I2C_Charac[I2C_speed].freq_max;
+
+	prev_error = ti2cspeed;
+
+	for (uint32_t count = 0; count < I2c_valid_timing_nbr; count++) {
+		/* tPRESC = (PRESC+1) x tI2CCLK*/
+		uint32_t tpresc = (I2c_valid_timing[count].presc + 1U) * ti2cclk;
```

Comment:
is this using tabs? The alignment seems to be different than in the rest of the file.

Please be consistent, 4 spaces should it be.